### DISCUSSION TITLE
Future prospects on ESP32 support

### DISCUSSION THREAD

**Discussion by eidetech** [slint-ui/slint#2534] [Q&A]
Just stumbled upon Slint, and I see that there is added some support for ESP32, specifically the Kaluga board. I was wondering what the future prospects on "general support" for ESP32 is, as I am developing an ESP32 application using LVGL for UI at the moment, in combination with the esp-idf framework.

I also noticed that Slint will support C++ and Rust, which I guess would mean that my esp-idf C codebase would need to be rewritten for use with Slint. So in order to avoid that, what does the future look like in regards to Slint support for Espressif SOC's? :)

**Comment by ogoffart** [comment]
Our approach is that there is no special need to support a board. As long as the architecture is supported by the rust compiler, it should work. The Slint API lets you draw in a any surface.  See the Rust documentation in https://docs.rs/slint/latest/slint/docs/mcu/index.html
We still don't have polished support for this API in C++ https://github.com/slint-ui/slint/issues/2136 , but once we have that, you could use your C code with the C++ API.
